### PR TITLE
Add crossdock-based integration tests for tracing propagation

### PR DESCRIPTION
**Pull request by yurishkuro**
There is a unit test that exercises Python-only permutations. The actual crossdock tests are not yet running as part of the CI, because I would like to do that in separate PRs.  Here's a small snippet of the results of running locally (with the corresponding Go PR https://github.com/uber/tchannel-go/pull/467)

```
✓ [trace] go→ (s1name=go s2encoding=json s2name=python s3encoding=thrift s3name=python sampled=true) (11/11 passed, 0/11 skipped)
   ├ ✓ ⇒ Trace ID should not be empty in S1(go)
   ├ ✓ ⇒ Trace ID must match in S1(go)
   ├ ✓ ⇒ Baggage must match in S1(go)
   ├ ✓ ⇒ Sampled must match in S1(go)
   ├ ✓ ⇒ Trace ID must match in S2(python:8081)
   ├ ✓ ⇒ Baggage must match in S2(python:8081)
   ├ ✓ ⇒ Sampled must match in S2(python:8081)
   ├ ✓ ⇒ Trace ID must match in S3(python:8081)
   ├ ✓ ⇒ Baggage must match in S3(python:8081)
   ├ ✓ ⇒ Sampled must match in S3(python:8081)
   └ ✓ ⇒ trace checks out
✓ [trace] go→ (s1name=go s2encoding=thrift s2name=python s3encoding=thrift s3name=python sampled=true) (11/11 passed, 0/11 skipped)
   ├ ✓ ⇒ Trace ID should not be empty in S1(go)
   ├ ✓ ⇒ Trace ID must match in S1(go)
   ├ ✓ ⇒ Baggage must match in S1(go)
   ├ ✓ ⇒ Sampled must match in S1(go)
   ├ ✓ ⇒ Trace ID must match in S2(python:8081)
   ├ ✓ ⇒ Baggage must match in S2(python:8081)
   ├ ✓ ⇒ Sampled must match in S2(python:8081)
   ├ ✓ ⇒ Trace ID must match in S3(python:8081)
   ├ ✓ ⇒ Baggage must match in S3(python:8081)
   ├ ✓ ⇒ Sampled must match in S3(python:8081)
   └ ✓ ⇒ trace checks out

352/352 passed (0/352 skipped)

Tests passed!
```

 * [x] HTTP server for health endpoint
 * [x] Start tchannel server that serves JSON endpoint
 * [x] Write test that hits JSON endpoint
 * [x] JSON endpoint must return observed span
 * [x] JSON endpoint must call downstream
 * [x] Write temp file with simple.thrift and create a service
 * [x] Add Thrift endpoint to do the same as JSON endpoint
 * [x] Update make file
 * [x] Test with Go image
 * [x] Docker Hub account
 * [x] Run crossdock test in Travis
 * [x] Pull real Go image
 * [x] Publish Python image

### CODE REVIEW

File: crossdock/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+    go:
+        image: tchannelhub/xdock-go:crossdock-in-travis
+        ports:
+            - "8080-8082"
+
+    python:
+        build:
+            context: ../.
+            dockerfile: crossdock/Dockerfile
```
Comment (breerly) [inline crossdock/docker-compose.yml]: this is a nice touch 👍

File: crossdock/setup_crossdock.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+from setuptools import setup, find_packages
+
+setup(
```
Comment (breerly) [inline crossdock/setup_crossdock.py]: any reason this file can't just be named `setup.py`?

File: requirements-test.txt [hunk 3]
```diff
@@ -39,7 +39,7 @@ thrift==0.9.2
 wrapt>=1.10,<1.11
 
 # OpenTracing reference implementation
-jaeger_client>=3,<4
+jaeger_client>=3.0.1,<4
```
Comment (breerly) [inline requirements-test.txt]: good pinning, ty sir 👍

File: tests/test_crossdock.py [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+        reporter=InMemoryReporter(),
+    )
+    try:
+        yield tracer
+    finally:
+        tracer.close()
+
+
+@pytest.mark.parametrize('s2_encoding,s3_encoding,sampled', [
+    ('json', 'json', True),
```
Comment (breerly) [inline tests/test_crossdock.py]: I wonder if enumerating all of these permutations could be done programmatically.

**Comment by breerly** [comment]
352 passed :shipit:

File: tests/test_crossdock.py (hunk 4 above)
Comment (yurishkuro) [inline tests/test_crossdock.py]: done

File: crossdock/setup_crossdock.py (hunk 2 above)
Comment (yurishkuro) [inline crossdock/setup_crossdock.py]: historical reasons, renamed

**Pull request merged**